src/optimizer/plugins/RemoveGeneratedTextRulesOptimizer.ts

```diff
@@ -19,31 +19,24 @@ export default {
 
   optimize(pkg: Package): void {
     [...pkg.profiles, ...pkg.extensions, ...pkg.valueSets, ...pkg.codeSystems].forEach(resource => {
-      const rulesToRemove: number[] = [];
       if (hasGeneratedText(resource)) {
-        resource.rules.forEach((rule: ExportableRule, i: number) => {
-          if (
-            rule instanceof ExportableCaretValueRule &&
-            rule.path === '' &&
-            rule.caretPath.match(/^text\./)
-          ) {
-            rulesToRemove.push(i);
-          }
-        });
+        resource.rules = (resource.rules as ExportableSdRule[]).filter(
```

Comment:
I guess TS is letting you say `resource.rules as ExportableSdRule[]`, but aren't some of the rules in `ExportableValueSet` and `ExportableCodeSystem` _not_ `ExportableSdRules`?  Wouldn't it be more correct to do something more generic like `resource.rules as fshrules.Rule[]`?

Reply:
Addressed in https://github.com/FHIR/GoFSH/pull/45/commits/a3cd779db79245595b46aabe2bd67c36e835a9fa, and also switched to using `remove`.